x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -1892,6 +1892,110 @@ public boolean exists(String field) {
         assertThat(Expressions.names(localSourceExec.output()), contains("languages", "min", "seen"));
     }
 
+    /**
+     * Expects
+     * intermediate plan
+     * LimitExec[500[INTEGER]]
+     * \_AggregateExec[[],[COUNT(emp_no{f}#6) AS c],FINAL,null]
+     *   \_ExchangeExec[[count{r}#16, seen{r}#17],true]
+     *     \_FragmentExec[filter=null, estimatedRowSize=0, fragment=[
+     * Aggregate[[],[COUNT(emp_no{f}#6) AS c]]
+     * \_Filter[emp_no{f}#6 > 10[INTEGER]]
+     *   \_EsRelation[test][_meta_field{f}#12, emp_no{f}#6, first_name{f}#7, ge..]]]
+     *
+     * and final plan is
+     * LimitExec[500[INTEGER]]
+     * \_AggregateExec[[],[COUNT(emp_no{f}#6) AS c],FINAL,8]
+     *   \_ExchangeExec[[count{r}#16, seen{r}#17],true]
+     *     \_LocalSourceExec[[count{r}#16, seen{r}#17],[LongVectorBlock[vector=ConstantLongVector[positions=1, value=0]]]]
+     */
+    public void testPartialAggFoldingOutput() {
+        var plan = physicalPlan("""
+              from test
+            | where emp_no > 10
+            | stats c = count(emp_no)
+            """);
+
+        var stats = statsForMissingField("emp_no");
+        var optimized = optimizedPlan(plan, stats);
+
+        var limit = as(optimized, LimitExec.class);
+        var agg = as(limit.child(), AggregateExec.class);
+        var exchange = as(agg.child(), ExchangeExec.class);
+        assertThat(Expressions.names(exchange.output()), contains("count", "seen"));
+        var source = as(exchange.child(), LocalSourceExec.class);
+        assertThat(Expressions.names(source.output()), contains("count", "seen"));
+    }
+
+    /**
+     * Checks that when the folding happens on the coordinator, the intermediate agg state
+     * are not used anymore.
+     *
+     * Expects
+     * LimitExec[10000[INTEGER]]
```

Comment:
Because, if there is a limit anywhere in the query, we add the default max. If there is no limit at all, we add the default 500. `Analyzer.AddImplicitLimit`